docker_diff.bzl

```diff
@@ -0,0 +1,69 @@
+# Copyright 2017 Google, Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def _impl(ctx):
+  """Implementation of docker_diff"""
+
+  image_location = ctx.executable.image.short_path
+  cd_binary_loction = ctx.executable._container_diff_tool.short_path
+  
+  content = "%s diff %s %s" % (cd_binary_loction, image_location, ctx.attr.diff_base)
```

Comment:
Could you make this a multi-line string, with a #!/bin/bash header and a set-e line?